test/e2e/tests/settings/show-native-as-main-balance.spec.js

```diff
@@ -0,0 +1,192 @@
+const { strict: assert } = require('assert');
+const { expect } = require('@playwright/test');
+const {
+  withFixtures,
+  defaultGanacheOptions,
+  logInWithBalanceValidation,
+  unlockWallet,
+  getEventPayloads,
+} = require('../../helpers');
+
+const FixtureBuilder = require('../../fixture-builder');
+
+async function mockSegment(mockServer) {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'Show native token as main balance' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+}
+
+describe('Settings: Show native token as main balance', function () {
+  it('Should show balance in crypto when toggle is on', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().withConversionRateDisabled().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver, ganacheServer }) => {
+        await logInWithBalanceValidation(driver, ganacheServer);
+
+        await driver.clickElement(
+          '[data-testid="account-overview__asset-tab"]',
+        );
+        const tokenValue = '25 ETH';
+        const tokenListAmount = await driver.findElement(
+          '[data-testid="multichain-token-list-item-value"]',
+        );
+        await driver.waitForNonEmptyElement(tokenListAmount);
+        assert.equal(await tokenListAmount.getText(), tokenValue);
+      },
+    );
+  });
+
+  it('Should show balance in fiat when toggle is OFF and not show popover twice', async function () {
```

Comment:
Do we need the toggle switch to test the pop-over only shows once?  Wondering if we could separate this into two tests on the `and` to toggle in one test and confirm the pop-up doesn't repeat in a second test.

Reply:
Hi Mike, thank you for your feedback 🙏  i refactored here: [563a536](https://github.com/MetaMask/metamask-extension/pull/27097/commits/563a536eafca604ffa62f560b62d0c5fef065f27)